bin/thingsvision.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+""" This is the command-line interface for the thingsvision toolbox """
+
+import argparse
+import torch
+import textwrap
+import sys
+
+parent_parser = argparse.ArgumentParser(
+    add_help=False,
+    prog="thingsvision",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description=textwrap.dedent(
+        """ This is the thingsvision CLI. It allows you to extract features from images using state-of-the-art neural networks for Computer Vision."""
+    ),
+)
+
+common_parser = argparse.ArgumentParser(parents=[parent_parser], add_help=False)
+common_parser.add_argument(
+    "--model-name",
+    type=str,
+    default="alexnet",
+    help="Name of the model to use for feature extraction.",
+)
+common_parser.add_argument(
+    "--source",
+    type=str,
+    default="torchvision",
+    choices=["torchvision", "keras", "timm", "custom"],
+    help="Source of the model to use for feature extraction. ",
+)
+common_parser.add_argument(
+    "--device",
+    type=str,
+    default="cpu",
+    choices=["cpu", "cuda"],
+    help="Device to use for the extractor.",
+)
+
+subparsers = parent_parser.add_subparsers(
+    title="Subcommands",
+    description="Valid subcommands are",
+    help="Additional help available is available within  each subcommand",
+    dest="command",
+)
+
+parser_model = subparsers.add_parser(
+    "show-model", description="Shows the model architecture", parents=[common_parser]
+)
+
+parser_extract = subparsers.add_parser(
+    "extract-features",
+    description="Extract features from images",
+    parents=[common_parser],
+)
+
+parser_extract.add_argument(
+    "--image-root",
+    type=str,
+    help="Path to directory containing images",
+    default="./images",
+)
+parser_extract.add_argument(
+    "--class-names",
+    type=str,
+    help="optional list of class names for class dataset",
+    default=None,
+)
+parser_extract.add_argument(
+    "--file-names",
+    type=str,
+    help="optional list of file names for class dataset",
+    default=None,
+)
+parser_extract.add_argument(
+    "--batch-size", type=int, default=32, help="Batch size used for feature extraction."
+)
+parser_extract.add_argument(
+    "--out-path",
+    type=str,
+    default="./features",
+    help="Path to directory where features should be stored.",
+)
+parser_extract.add_argument(
+    "--flatten-acts", action="store_true", help="Flatten activations before saving."
+)
+parser_extract.add_argument(
+    "--model-parameters",
+    type=str,
+    default=None,
+    help="Clip vision model has to be defined in the model_parameters dictionary",
+)
+parser_extract.add_argument(
+    "--module-name",
+    type=str,
+    default="features.10",
+    help="Name of the module to use for feature extraction.",
+)
```

Comment:
Not sure I like a default module name here. I was also on the fence about having AlexNet as the default model, but I can actually accept that. I think if you're using `thingsvision` you have an idea of what you want anyway and probably should know that you need to specify a specific layer. I think  it's too easy to specifiy a different model and maybe not specify a layer (that doesn't even have a `features.`10`). I think this is okay for the moment, but probably for next version, maybe figuring out a better way to work with defaults.